src/controllers/stringsController.js

```diff
@@ -0,0 +1,19 @@
+const { sayHello, uppercase, lowercase, firstCharacters } = require('../lib/strings');
+
+exports.strings_sayhello = (req, res) => {
+  res.status(200).json({ result: sayHello(req.params.string) });
+};
+
+exports.strings_uppercase = (req, res) => {
+  res.status(200).json({ result: uppercase(req.params.string) });
+};
+
+exports.strings_lowercase = (req, res) => {
+  res.status(200).json({ result: lowercase(req.params.string) });
+};
+
+exports.strings_firstCharacters = (req, res) => {
+  const length = req.query.length || 1;
```

Comment:
Again, additional info here -> Have you seen lodash before? There is a method called get which is good for when you're relying on nested object data. For example, if req is undefined then you'll get an error can't read query of undefined. If req is defined but query is not, you will get an error cannot read length of undefined. Get is handy because it means you can use that function instead of writing req && req.query && req.query.length to check the full path is defined.

Reply:
I'll need to look that up, haven't come across lodash yet, sounds handy :)